src/core/io/src/4C_io_input_file.cpp

```diff
@@ -474,17 +504,34 @@ namespace Core::IO
       content.set_up_fragments();
     }
 
-    // All content has been read. Now validate. In the first iteration of this new feature,
-    // we only validate the section names, not the content.
+    // All content has been read. Now validate.
     if (Core::Communication::my_mpi_rank(pimpl_->comm_) == 0)
     {
-      for (const auto& [section_name, content] : pimpl_->content_by_section_)
+      InputParameterContainer version_data;
+      match_section(input_version_key, version_data);
+
+      if (auto opt_version = version_data.get<std::optional<std::string>>(input_version_key);
+          opt_version.has_value())
       {
-        if (!pimpl_->is_section_known(section_name))
+        if (*opt_version != VersionControl::input_version)
+          std::cout << std::format(
+              "Warning: version mismatch: "
+              "The input file requires input version {}, but the input version of 4C is {}. "
+              "Your input file may be incompatible with 4C.\n",
+              *opt_version, VersionControl::input_version);
```

Comment:
Do you want to mention that the input version is experimental?
```suggestion
              "Warning: version mismatch for experimental input version: "
              "The input file requires input version {}, but the input version of 4C is {}. "
              "Your input file may be incompatible with 4C.\n",
              *opt_version, VersionControl::input_version);
```